### ISSUE TITLE
Add CVar `mp_plant_c4_delay`

### ISSUE THREAD

**Issue by wopox1337** [s1lentq/ReGameDLL_CS#698]
```cpp
mp_plant_c4_delay <float> 
```
And don't forget to implement API member in `CSPlayer()->*`

**Comment by wopox1337** [comment]
I completely agree with you on the point of misunderstanding the player and the missing necessary message. Unfortunately, I have not yet found a solution to correct this situation. In search...

Regarding API - CVar should not regulate (influence, interfere with) the work of API. That's why the logic is implemented in a simpler way.

**Comment by StevenKal** [comment]
As I said on the other topic #694, I made a draft of it & by respecting your actual desires.
Meantime, I can see a problem with actual (more a limit in fact), which, the fact we can not fully customize the "delay" per bombsite/anywhere.
In other terms, with actual design, if you want to plant anywhere, but only from 30 seconds after round start, but also plant on bomb site, but without delay (like if you reach the site in 20 seconds), you can not, you must wait those famous 30 seconds of "mp_plant_c4_delay".
So to allow full control of this, we must have CVars like:
mp_plant_c4_site_delay -1 (default value, -1 for limit, 0 to disable planting haha!, <delay> for our delay).
mp_plant_c4_anywhere_delay 0 (default value, -1 for limit, 0 to disable planting haha!, <delay> for our delay).

So the question is, should I try to change the whole code to handle like I suggest, or should I keep the actual behavior wanted?
That's you who can valid the features, so that's you who decide! Hehe!

**Comment by StevenKal** [comment]
Hi!
I am still waiting for your decision regarding my suggestion right above, because this last will provide more customization (check explanations).
Besides, this will not be a problem for me to rework my actual code to handle it like I suggested, I just do not want to code this if in case you disagree & refuse to merge such design (no need I waste time huh!).

However, I have a finalized & fully workable code [here](https://github.com/StevenKal/ReGameDLL_CS/pull/6), get the "Lara Croft's artifacts" related to the PR I have opened on my repository to test.
I will just need to add the CVar in readme & game.cfg files, that's all!

**Comment by StevenKal** [comment]
zZzZz!
I just would like to know if in case I work on a code with "mp_plant_c4_site_delay" & "mp_plant_c4_anywhere_delay" to behave like I suggested on my post of 2021-10-30, if you will be favorable to merge it if I submit, or totally against.
This is as simple as that!
My first guessing is "you will be against", but a confirmation will be appreciated! Thanks!

**Comment by wopox1337** [comment]
Thank you for your explanation. Yes, your arguments about the gameplay part are quite logical, as I said before - the "delay" adjustment is a necessity. Also, control in API without depending on CVar itself - looks better than now.

**Comment by StevenKal** [comment]
Oh! Cute wopox is there! I though you were gone into oblivion! Hehehe!
I can see a "up" in my post of the 2021-10-30, but this does not really tell me if you are fully O.K. if I work & submit a new code like I have explained with the new CVars "mp_plant_c4_<site|anywhere>_delay", etc.?
Just say "yes" if you are O.K., or "no" if you are not! Thanks in advance!

**Comment by wopox1337** [comment]
I don't have a lot of time to do these projects. But, sometimes I will be able to help with some things. I'M OK.

Yes, you can suggest a PR and we'll look into it (review).
